scripts/zones/Port_Jeuno/npcs/Squintrox_Dryeyes.lua

```diff
@@ -3,117 +3,534 @@
+            player:messageSpecial(ID.text.KEYITEM_OBTAINED, ki)
+        else
+            player:messageSpecial(ID.text.DRYEYES_3, ki)
+        end
+    else
+        player:showText(npc, ID.text.GET_LOST)
     end
 end
 
 entity.onTrigger = function(player, npc)
-    if
-        xi.settings.main.ENABLE_ACP == 0 and
-        xi.settings.main.ENABLE_AMK == 0 and
-        xi.settings.main.ENABLE_ASA == 0
-    then
+    local now          = os.time()
+    local nextMidnight = getMidnight()
+    local finishedACP  = player:getCurrentMission(xi.mission.log_id.ACP) == xi.mission.id.acp.A_CRYSTALLINE_PROPHECY_FIN
+    local finishedAMK  = player:getCurrentMission(xi.mission.log_id.AMK) == xi.mission.id.amk.A_MOOGLE_KUPO_DETAT_FIN
+    local finishedASA  = player:getCurrentMission(xi.mission.log_id.ASA) == xi.mission.id.asa.A_SHANTOTTO_ASCENSION_FIN
+
+    -- Reminder that "True" means the option should be excluded from the player's menu
+    local arg1 =
+        ((finishedACP                                              or
+            player:hasKeyItem(xi.ki.CRIMSON_KEY))                  and
+                2 or 0) +
+        ((finishedACP                                              or
+            player:hasKeyItem(xi.ki.VIRIDIAN_KEY))                 and
+                4 or 0) +
+        ((finishedAMK                                              or
+            player:hasKeyItem(xi.ki.WHITE_CORAL_KEY))              and
+                8 or 0) +
+        ((finishedAMK                                              or
+            player:hasKeyItem(xi.ki.BLUE_CORAL_KEY))               and
+                16 or 0) +
+        ((finishedAMK                                              or
+            player:hasKeyItem(xi.ki.BLACK_CORAL_KEY))              and
+                32 or 0) +
+        ((finishedASA                                              or
+            player:hasKeyItem(xi.ki.MOOGLE_KEY)                    and
+            now >= player:getCharVar('LastMoogleKey'))             and
+                64 or 0) +
+        ((finishedASA                                              or
+            player:hasKeyItem(xi.ki.BIRD_KEY)                      and
+            now >= player:getCharVar('LastBirdKey'))               and
+                128 or 0) +
+        ((finishedASA                                              or
+            player:hasKeyItem(xi.ki.BOMB_KEY)                      and
+            now >= player:getCharVar('LastBombKey'))               and
+                256 or 0)
+
+    -- matches none of the above
+    if arg1 == 510 then
         player:showText(npc, ID.text.GET_LOST)
     else
-        player:startEvent(323)
+        player:startEvent(323, arg1)
     end
 end
 
+-- validation of prereqs done here, event option updated based on checks to be used in onEventFinish
 entity.onEventUpdate = function(player, csid, option, npc)
+    if csid == 323 then
+        if option == 100 then -- Viridian Key
```

Comment:
Each of the blocks in here are quite unique, and take up quite varying amounts of space - I'd probably break all of them out into their own functions